snapshot/snapshotter.go

```diff
@@ -14,23 +14,36 @@ type Kind uint8
 
 // definitions of snapshot kinds
 const (
-	KindView Kind = iota + 1
+	KindUnkown Kind = iota
+	KindView
 	KindActive
 	KindCommitted
 )
 
-var (
-	kindStrings = map[Kind]string{
-		KindView:      "view",
-		KindActive:    "active",
-		KindCommitted: "committed",
+func ParseKind(s string) Kind {
+	s = strings.ToLower(s)
+	switch s {
+	case "view":
+		return KindView
+	case "active":
+		return KindActive
+	case "committed":
+		return KindCommitted
 	}
-)
+
+	return KindUnkown
+}
 
 func (k Kind) String() string {
-	if s, ok := kindStrings[k]; ok {
-		return s
+	switch k {
+	case KindView:
+		return "View"
```

Comment:
The casing has changed here

Reply:
Na, we just go with this casing then flatten the case when parsing.